### PR TITLE
Extend country input field functionality

### PR DESCRIPTION
**Pull request by moiikana**
... in donation and membership form.

https://phabricator.wikimedia.org/T247227

### CODE REVIEW

File: skins/laika/src/components/shared/Postal.vue [hunk 1]
```diff
@@ -42,45 +42,78 @@
...
 		<label for="country" class="subtitle">{{ $t( 'donation_form_country_label' ) }}</label>
-		<b-field>
-			<b-select
+		<b-field :type="{ 'is-danger': showError.country }">
+			<b-autocomplete
 					class="is-form-input"
-					v-model="formData.country.value"
-					id="country"
+					field="countryFullName"
+					placeholder="z. B. Deutschland"
```
Comment (gbirke) [inline skins/laika/src/components/shared/Postal.vue:52]: This needs to be translatable

File: skins/laika/src/components/shared/Postal.vue [hunk 2]
```diff
@@ -42,45 +42,78 @@
...
+					v-model="countryInput"
 					name="country"
-					@blur="$emit('field-changed', 'country')">
-				<option v-for="(countryCode, index) in countries"
-						:value="countryCode"
-						:key="index">{{ $t('donation_form_country_option_' + countryCode ) }}
-				</option>
-			</b-select>
+					id="country"
```
Comment (gbirke) [inline skins/laika/src/components/shared/Postal.vue:55]: `id` is probably a vestige that can be removed

File: skins/laika/src/components/shared/Postal.vue [hunk 3]
```diff
@@ -92,6 +125,15 @@ export default Vue.extend( {
...
 		},
+
+		filteredCountries(): Array<Country> {
+			return this.countries.filter( ( countryOption: Country ) => {
+				return countryOption.countryFullName
+					.toString()
+					.toLowerCase()
+					.indexOf( this.$data.countryInput.toLowerCase() ) >= 0;
+			} );
+		},
```
Comment (gbirke) [inline skins/laika/src/components/shared/Postal.vue:136]: Why is this method necessary? I'd expect the autocomplete to whittle down the full list based on the current input.

Comment (moiikana) [reply skins/laika/src/components/shared/Postal.vue:136]: buefy's autocomplete docs demonstrate it like that. we need our own filter

File: skins/laika/src/components/shared/Postal.vue (hunk 2 above)
Comment (Abban) [reply skins/laika/src/components/shared/Postal.vue:55]: It's used for the label targeting

**Pull request merged**